ceph_iscsi_config/client.py

```diff
@@ -199,7 +201,7 @@ def define_client(self):
         # The configuration only has one active tpg, so pick that one for any
         # acl definitions
         for tpg in r.tpgs:
-            if tpg.enable:
+            if tpg.enable and tpg.parent_target.wwn == self.target_iqn:
```

Comment:
You can just detect this in the "for each tpg in target.tpgs:" loop too.

Reply:
Fixed